bemuse/src/app/entities/Options.js

```diff
@@ -25,15 +25,15 @@ export const changeKeyMapping = (mode, key, keyCode) =>
 // Play mode
 export const playMode = (state) => state['player.P1.mode']
 export const changePlayMode = (mode) =>
-  u({
-    'player.P1.mode': mode,
-    'player.P1.panel': (panel) =>
-      panel === '3d' && mode !== 'KB' ? 'center' : panel,
+  produce((draft) => {
+    draft['player.P1'.mode] = mode
+    draft['player.P1.panel'] = (panel) =>
+      panel === '3d' && mode !== 'KB' ? 'center' : panel;
```

Comment:
It is not correct to assign a function to a draft. When you see a function used in `updeep`, it means updating the value based on the previous value. The equivalent on `immer` looks like this:

```suggestion
    const panel = draft['player.P1.panel']
    draft['player.P1.panel'] = panel === '3d' && mode !== 'KB' ? 'center' : panel
```